### ISSUE TITLE
Autocomplete in html files?

### ISSUE THREAD

**Issue by jeffreymeng** [TypeStrong/atom-typescript#1355]
Is there a way for it to autocomplete typescript in `.html` typescript files?

So in webstorm(trial), it works like this:
![](https://i.stack.imgur.com/mszt0.png)

In atom `.html` files, however, there are no autocomplete functions.
![](https://i.stack.imgur.com/8LR1t.png)

This is the same question as the one on [Stack OVerflow](https://stackoverflow.com/questions/39557728/code-completion-autocomplete-angular-typescript-atom-in-html/46615135#46615135), but has no answers yet.

Thank You

**Comment by jonyeezs** [comment]
This is a duplicate of #1278.

This pkg focuses mainly on ts files. I don't think this would be a typescript feature?

I've proposed, in that linked issue, to have atom-typescript to open an API so we can have plugins into the IDE.

And hopefully with [enabling the TS language service plugin](https://github.com/Microsoft/TypeScript/wiki/Writing-a-Language-Service-Plugin#enabling-a-plugin) and a package that can utilize this atom-typescript plugin would make this work.

**Comment by jonyeezs** [comment]
or maybe have `atom-typscript` configurable to install plugins and expand its acceptable file types.

....hmmm maybe that would be an easy way...

**Comment by lierdakil** [comment]
FWIW, making accepted file types configurable is trivial. Grammars are defined here: https://github.com/TypeStrong/atom-typescript/blob/b4fb237e1934ba1e7373dbdc1c7c3aeba0676130/lib/main/atom/utils/atom.ts#L25
File extensions (because we need to detect typescript files that aren't open in editors necessarily) are a couple lines below:
https://github.com/TypeStrong/atom-typescript/blob/b4fb237e1934ba1e7373dbdc1c7c3aeba0676130/lib/main/atom/utils/atom.ts#L29

Changing those hard-coded arrays to `atom.config.get('atom-typescript.typescriptGrammars')` for example should be the long and short of it.

However, I'm really not sure it'll be enough.

**Comment by jonyeezs** [comment]
@lierdakil yeah that would be the easy part. I'm hoping that including a service language plugin that would support html would just work.

I'll try experiment with this

**Comment by admosity** [comment]
Maybe a combination of the angular language service: https://angular.io/guide/language-service

And: https://github.com/atom/atom-languageclient

For targeting just angular html.

**Comment by DerZyklop** [comment]
I am still interested in this. If anyone has a workaround or hints how to get this, please post it here 
✌

**Comment by arnotes** [comment]
im also interested

**Comment by DerZyklop** [comment]
@jonyeezs @arnotes @jeffreymeng  Does it make sense to open a new Ticket? #1355 and #1278 got closed by bot, but are still a thing.

**Comment by lierdakil** [comment]
@DerZyklop, it does not. The issue of supporting HTML specifically is out of scope frankly. Here's what you should do:

1. Write/find a language service plug-in for html support (some say angular language service might work)
2. Install & enable this plug-in in your `tsconfig.json`.
3. Add html to the list of TypeScript files/grammars in Atom-TypeScript settings.

Stuff should work. If it doesn't, open an issue. But don't open an issue if you're stuck on step (1) or (2), since those are not directly related to this project and this bug tracker is not a general TypeScript help forum.

**Comment by XephyLon** [comment]
Opened an Angular project, added the LS plugin in
```json
{
  "compileOnSave": false,
  "plugins": [{
    "name": "@angular/language-service"
  }],
}
```
Then added `.component.html` and `.html` to the supported files
![image](https://user-images.githubusercontent.com/37638392/95873856-95767a00-0d70-11eb-871c-540e7a2e9d85.png)

Restarted Atom and nothing works. Was hoping for any sort of error to pop up at least.

Please let me know if I'd done something wrong or if that's just how it should be.

**Comment by XephyLon** [comment]
OK here's an update.
![image](https://user-images.githubusercontent.com/37638392/95877575-8a254d80-0d74-11eb-9d21-32068fdf20f2.png)

**Comment by lierdakil** [comment]
Welp, I've spent a couple hours fiddling with this, and found this comment in the bowels of Angular's language service plug-in:
```
        //    Plugin augments TS language service to provide additional Angular
        //    information. This only works with inline templates and is meant to be
        //    used as a local plugin (configured via tsconfig.json)
```
so this apparently won't work.

Note, if targeting Angular specifically, you probably don't actually want to use Atom-TypeScript at all. Instead, you should perhaps look towards integrating `@angular/language-server` with Atom via `atom-languageclient`. A basic integration should be more or less straightforward. I wouldn't mind whipping up a proof of concept provided I won't have to maintain that 
-- have my hands full as it is.

**Comment by lierdakil** [comment]
Okay, I'm at a loss. So I've slapped together a prototype https://github.com/lierdakil/atom-ide-angular. I've made a few such packages and seen a lot more, so it shouldn't have been an issue. And it's fine, the language server works, it accepts requests, it prints debug messages to console, all's fine. The only problem is all replies are empty: it returns empty diagnostics for obviously broken files, it returns empty completions, empty hover tooltips, so basically it "works", but does nothing. I don't have any more time to dedicate to this, and considering I don't know first thing about Angular, maybe I'm just missing something stupid. If anyone wants to try to figure this out, feel free to fork the repo, hack on the code, release the package, etc.
